tools/gcp_kms_secrets.py

```diff
@@ -0,0 +1,406 @@
+        print("An exception occurred encrypting secrets:")
+        print("{}\n".format(err))
+
+    # Write secrets into new terraform.tfvars file
+    kms_cryptokey_id = client.crypto_key_path_path(project_id, location, key_ring_id, crypto_key_id)
+    write_new_tfvars(tfvars_path, secrets, kms_cryptokey_id)
+
+
+# Prompt user for deployment type (multi-region or single) and returns the path to the tfvars
+def get_tfvars_path():
+    validInput = False
+    deployment = None
+
+    while not validInput:
+        deployment = input("Enter 0, 1, or 2 for the deployment type:\n"
+                           "[0] for single-connector\n"
+                           "[1] for multi-region\n"
+                           "[2] for dc-only\n")
+        
+        if deployment == "0" or deployment == "1" or deployment == "2":
+            validInput = True
+
+    # Read the corresponding .tfvars file based on user selection
+    switcher = {
+        "0": "../deployments/gcp/single-connector/terraform.tfvars",
+        "1": "../deployments/gcp/multi-region/terraform.tfvars",
+        "2": "../deployments/gcp/dc-only/terraform.tfvars"
+    }
+
+    return switcher.get(deployment)
+
+
+# Collect the required secrets for the appropriate deployment from the tfvars file
+def get_tfvars_secrets(tfvars_dict, tfvars_path):
+    secrets = {
+        "dc_admin_password"           : tfvars_dict.get("dc_admin_password"),
+        "safe_mode_admin_password"    : tfvars_dict.get("safe_mode_admin_password"),
+        "ad_service_account_password" : tfvars_dict.get("ad_service_account_password")
+    }
+    
+    if "single-connector" in tfvars_path or "multi-region" in tfvars_path:
+        secrets.update(pcoip_registration_code = tfvars_dict.get("pcoip_registration_code"))
+
+        global CAM_CREDENTIALS_FILE
+        CAM_CREDENTIALS_FILE = tfvars_dict.get("cam_credentials_file")
+
+    return secrets
+
+
+# List KMS key rings
+def list_key_rings(client, project_id, location):
+    parent        = client.location_path(project_id, location)
+    response      = client.list_key_rings(parent)
+    response_list = list(response)
+
+    if len(response_list) > 0:
+        print("Key rings in project:")
+        for key_ring in response_list:
+            print("{}\n".format(key_ring.name))
+    else:
+        print("No key rings found.\n")
+
+
+# Read terraform.tfvars for user provided configurations
+def read_terraform_tfvars(tfvars_file):
+    tf_data = {}
+
+    with open(tfvars_file, 'r') as f:
+        for line in f:
+            if line[0] in ('#', '\n'):
+                continue
+            
+            # Split using the first delimiter
+            key, value = map(str.strip, line.split('=', 1))
+            tf_data[key] = value.replace("\"", "")
+
+    return tf_data
+
+
+# Write the new tfvars file using encrypted secrets
+def write_new_tfvars(tfvars_file, secrets, kms_cryptokey_id):
+    kms_resource_list  = kms_cryptokey_id.split('/')
```

Comment:
best if you can give an example of why "5" and "7" are chosen, so reader doesn't have to google what the string is supposed to look like.